sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_feature_set/source_metadata_schema.py

```diff
@@ -11,14 +11,17 @@
 from azure.ai.ml._schema.core.schema import PatchedSchemaMeta
 
 from .delay_metadata_schema import DelayMetadataSchema
+from .source_process_code_metadata_schema import SourceProcessCodeSchema
 from .timestamp_column_metadata_schema import TimestampColumnMetadataSchema
 
 
 class SourceMetadataSchema(metaclass=PatchedSchemaMeta):
     type = fields.Str(required=True)
-    path = fields.Str(required=True)
+    path = fields.Str(required=False)
     timestamp_column = fields.Nested(TimestampColumnMetadataSchema, required=True)
     source_delay = fields.Nested(DelayMetadataSchema, required=False)
+    source_process_code = fields.Nested(SourceProcessCodeSchema, load_only=True, required=False)
```

Comment:
If set to load_only, these will not be visible to user if they try to print(featureset)?

Reply:
Even for now, user will not see feature set spec when print(featureset). User can print feature set spec in data plane, in control plane, the schema is mainly for validation.